src/Resource/Collection.js

```diff
@@ -0,0 +1,101 @@
+import ParameterBag from './ParameterBag';
+
+export default class Collection {
+  /**
+   * @param {String} name
+   * @param {HyralRepository} repository
+   */
+  constructor(name, repository) {
+    /** @private */
+    this.name = name;
+
+    /** @private */
+    this.repository = repository;
+
+    /** @private */
+    this.metadata = {
+      loading: false,
+      loaded: false,
+      parameterBag: null,
+      paging: {
+        count: null,
+        pages: null,
+      },
+    };
+
+    /** @private */
+    this.loadingPromise = null;
+
+    /** @private */
+    this.data = {
+      items: [],
+    };
+  }
+
+  get parameterBag() {
+    if (!this.metadata.parameterBag) {
+      this.metadata.parameterBag = new ParameterBag();
+    }
+
+    return this.metadata.parameterBag;
+  }
+
+  set parameterBag(parameterBag) {
+    this.metadata.parameterBag = parameterBag;
+  }
+
+  /**
+   * @returns {number}
+   */
+  get length() {
+    return this.metadata.paging.count || 0;
+  }
+
+  /**
+   * @returns {number}
+   */
+  get pages() {
+    return this.metadata.paging.pages || 0;
+  }
+
+  /**
+   * @returns {boolean}
+   */
+  get isLoading() {
+    return this.metadata.loading;
+  }
+
+  /**
+   * @returns {boolean}
+   */
+  get isLoaded() {
+    return this.metadata.loaded;
+  }
+
+  /**
+   * @returns {Array}
+   */
+  get items() {
+    return this.data.items;
+  }
+
+  /**
+   * @param {Array} items
+   */
+  set items(items) {
```

Comment:
Ik twijfel eigenlijk of we een setter van de items willen hebben. Het voelt alsof dit alleen een getter zou moeten zijn.
Als er dan wel een setter voor is: zou je dan niet loaded op true zetten? (Al is dit strict genomen ook niet waar, want je hebt eigenlijk niks geladen).

Reply:
mag weg idd